tools/serverpod_cli/test/generator/dart/server_code_generator/database_repository_class_additions_test.dart

```diff
@@ -0,0 +1,455 @@
+import 'package:analyzer/dart/analysis/utilities.dart';
+import 'package:serverpod_cli/src/generator/dart/server_code_generator.dart';
+import 'package:serverpod_cli/src/test_util/builders/class_definition_builder.dart';
+import 'package:serverpod_cli/src/test_util/builders/generator_config_builder.dart';
+import 'package:serverpod_cli/src/test_util/compilation_unit_helpers.dart';
+import 'package:test/test.dart';
+import 'package:path/path.dart' as path;
+
+const projectName = 'example_project';
+final config = GeneratorConfigBuilder().withName(projectName).build();
+const generator = DartServerCodeGenerator();
+
+void main() {
```

Comment:
Question: What does `additions` mean in the test name?

Reply:
I think I was refering to additive props `attach` and `detach`

Reply:
These are all in the same class right? We might need to sync bit on how we group the tests.

Reply:
Attach and Detach are their own classes in the format of `${className}AttachRowRepository` and `${className}DetachRowRepository`. These classes are the constructed inside the `${className}Repository` as the fields `attachRow = ${className}AttachRowRepository._();` and `detachRow = ${className}DetachRowRepository._();`

The tests here tests the stand alone classes.

Reply:
Aah ok! Then we are good.

Would it be worth splitting them into their own files and naming the file after the class like we do for the rest?

Reply:
It might be yeah! But I also think I can do that when I introduce the batch operations will be a quite natural fit for that PR I think!